code/game/objects/items/devices/PDA/apps/apps.dm

```diff
@@ -60,6 +60,8 @@
 	U.set_machine(pda_device)
 
 /datum/pda_app/proc/refresh_pda()
+	if(!pda_device)
```

Comment:
when can this happen?

Reply:
When detonating your own PDA with a detomatix and this code, it'll runtime and complain about "null.attack_self()" without this